src/printer.js

```diff
@@ -2206,20 +2221,32 @@ function printJSXElement(path, options, print) {
   let forcedBreak = false;
 
   // Trim trailing whitespace, recording if there was a hardline
+  let numTrailing = 0;
```

Comment:
My hunch is it may have to do with the newlines belonging to different groups...